File: 7_kyu/The_fusc_function_Part_1.go
package kata

func Fusc(n int) int {
	for n > 0 && n % 2 == 0 {
		n /= 2
	}
	if n < 2 {
		return n
	} else {
		n /= 2
		return Fusc(n) + Fusc(n+1)
	}
}

File: 7_kyu/The_fusc_function_Part_1.rs
// The fusc function as specified.
fn fusc(n: u32) -> u32 {
    match n {
        0 | 1 => n,
        _ if n & 1 == 0 => fusc(n / 2),
        _ => fusc(n / 2) + fusc(n / 2 + 1),
    }
}
